src/middlewares/error.middleware.ts

```diff
@@ -0,0 +1,35 @@
+import { Request, Response, NextFunction } from 'express'
+import { HttpException } from '@exceptions/HttpException'
+
+/**
+ * Custom error handler to standardize error objects returned to
+ * the client
+ *
+ * @param err Error caught by Express.js
+ * @param req Request object provided by Express
+ * @param res Response object provided by Express
+ * @param next NextFunction function provided by Express
+ */
+function errorHandler(
+  err: TypeError | HttpException,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  let customError = err
+
+  if (!(err instanceof HttpException)) {
+    customError = new HttpException(
+      'Oh no, this is embarrasing. We are having troubles my friend'
+    )
+  }
+  console.log(customError)
+
+  // we are not using the next function to prevent from triggering
+  // the default error-handler. However, make sure you are sending a
+  // response to client to prevent memory leaks in case you decide to
+  // NOT use, like in this example, the NextFunction .i.e., next(new Error())
```

Comment:
I am new to node, is the memory leak caused by the client waiting for a response (which will not be sent in the event that we do not use `next` to pass the error to other middlewares AND no response is sent)?